### PR TITLE
ENH: support StringMethods index and rindex

### PR DESCRIPTION
**Pull request by mortada**
as a part of https://github.com/pydata/pandas/issues/9111

@sinhrks @jreback

### CODE REVIEW

**Comment by jreback** [comment]
are there any other routines that have an ``array`` as a return type?

**Comment by mortada** [comment]
@jreback good catch, return type of `array` is used in the old docstrings, I'll update it

I also added to the release note

**Comment by mortada** [comment]
@jreback do you think this can make 0.16.1? seems like it'd be nice to get all the string methods in

File: doc/source/whatsnew/v0.16.1.txt [hunk 1]
```diff
@@ -39,6 +39,7 @@ Enhancements
      Timestamp('2014-08-01 07:00') + BusinessHour()
      Timestamp('2014-08-01 16:30') + BusinessHour()
 
+- Added ``StringMethods.index()`` and ``rindex`` which behave the same as standard ``str`` (:issue:`10045`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.1.txt]: can u move this in the whatsnew with all the other steing changes

**Comment by mortada** [comment]
@jreback i updated the release note

the only other PR remaining is for `str.translate()` here: https://github.com/pydata/pandas/pull/10052/

**Comment by jreback** [comment]
was also passing before, right?

**Comment by mortada** [comment]
yes this was passing too, just changing docs

**Pull request merged**